### ISSUE TITLE
sstable2json cannot read sstables generated by Scylla 1.3

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#1566]
HEAD = d7f8ce77223258507b8a16d21d0deb15643b6f64

Works with Scylla 1.2.3.

```
Exception in thread "main" java.lang.IndexOutOfBoundsException
	at java.nio.ByteBuffer.wrap(ByteBuffer.java:375)
	at org.apache.cassandra.db.composites.AbstractCompoundCellNameType$CompositeDeserializer.deserializeOne(AbstractCompoundCellNameType.java:233)
	at org.apache.cassandra.db.composites.AbstractCompoundCellNameType$CompositeDeserializer.hasComponent(AbstractCompoundCellNameType.java:211)
	at org.apache.cassandra.db.composites.AbstractCompoundCellNameType$CompositeDeserializer.compareNextTo(AbstractCompoundCellNameType.java:177)
	at org.apache.cassandra.db.AtomDeserializer.compareNextTo(AtomDeserializer.java:85)
	at org.apache.cassandra.db.columniterator.IndexedSliceReader$IndexedBlockFetcher.getNextBlock(IndexedSliceReader.java:396)
	at org.apache.cassandra.db.columniterator.IndexedSliceReader$IndexedBlockFetcher.fetchMoreData(IndexedSliceReader.java:350)
	at org.apache.cassandra.db.columniterator.IndexedSliceReader.computeNext(IndexedSliceReader.java:142)
	at org.apache.cassandra.db.columniterator.IndexedSliceReader.computeNext(IndexedSliceReader.java:44)
	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
	at org.apache.cassandra.db.columniterator.SSTableSliceIterator.hasNext(SSTableSliceIterator.java:82)
	at org.apache.cassandra.db.filter.QueryFilter$2.getNext(QueryFilter.java:173)
	at org.apache.cassandra.db.filter.QueryFilter$2.hasNext(QueryFilter.java:156)
	at org.apache.cassandra.db.filter.SliceQueryFilter.collectReducedColumns(SliceQueryFilter.java:264)
	at org.apache.cassandra.db.filter.QueryFilter.collateColumns(QueryFilter.java:108)
	at org.apache.cassandra.db.filter.QueryFilter.collateOnDiskAtom(QueryFilter.java:82)
	at org.apache.cassandra.db.filter.QueryFilter.collateOnDiskAtom(QueryFilter.java:69)
	at org.apache.cassandra.db.CollationController.collectAllData(CollationController.java:314)
	at org.apache.cassandra.db.CollationController.getTopLevelColumns(CollationController.java:65)
	at org.apache.cassandra.db.ColumnFamilyStore.getTopLevelColumns(ColumnFamilyStore.java:1967)
	at org.apache.cassandra.db.ColumnFamilyStore.getColumnFamily(ColumnFamilyStore.java:1810)
	at org.apache.cassandra.db.ColumnFamilyStore.getColumnFamily(ColumnFamilyStore.java:1613)
	at org.apache.cassandra.db.SystemKeyspace.readSchemaRow(SystemKeyspace.java:870)
	at org.apache.cassandra.config.CFMetaData.fromSchema(CFMetaData.java:1860)
	at org.apache.cassandra.config.KSMetaData.deserializeColumnFamilies(KSMetaData.java:321)
	at org.apache.cassandra.config.KSMetaData.fromSchema(KSMetaData.java:302)
	at org.apache.cassandra.db.DefsTables.loadFromKeyspace(DefsTables.java:133)
	at org.apache.cassandra.config.DatabaseDescriptor.loadSchemas(DatabaseDescriptor.java:696)
	at org.apache.cassandra.tools.SSTableExport.main(SSTableExport.java:422)
```

Copy of the sstable: [booba2.zip](https://github.com/scylladb/scylla/files/411067/booba2.zip)

```
CREATE TABLE ks1.booba2 (
    pk int,
    ck text,
    s1 int static,
    v int,
    PRIMARY KEY (pk, ck)
) WITH CLUSTERING ORDER BY (ck ASC)
    AND bloom_filter_fp_chance = 0.01
    AND caching = '{"keys":"ALL","rows_per_partition":"ALL"}'
    AND comment = ''
    AND compaction = {'class': 'SizeTieredCompactionStrategy'}
    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';

```

**Comment by slivne** [comment]
I gather that booba2.zip
<https://github.com/scylladb/scylla/files/411067/booba2.zip> files are
created corrupt ...

can you also include the insert/delete statements used to generate the data

**Comment by tgrabiec** [comment]
Looks like it's fixed by f1c2481
